herddb-core/src/test/java/herddb/checkpoint/CheckpointCommandTest.java

```diff
@@ -0,0 +1,121 @@
+                execute(manager, "INSERT INTO tblspace1.tsql (k1,n1 ,s1) values (?,?,?)", Arrays.asList(i, 1, "b"));
+            }
+            execute(manager, "checkpoint " + wrongTableSpace, Collections.emptyList());
+        }
+    }
+
+    @Test(expected = herddb.model.TableSpaceDoesNotExistException.class)
+    public void checkpointCommandCaseInsensitiveFailureTest() throws Exception {
+        try (DBManager manager = new DBManager("localhost", new MemoryMetadataStorageManager(), new MemoryDataStorageManager(), new MemoryCommitLogManager(), null, null)) {
+            manager.start();
+            assertTrue(manager.waitForBootOfLocalTablespaces(10000));
+            execute(manager, "CREATE TABLESPACE 'tblspace1'", Collections.emptyList());
+            manager.waitForTablespace("tblspace1", 10000);
+            execute(manager, "CREATE TABLE tblspace1.tsql (k1 string primary key,n1 int,s1 string)", Collections.emptyList());
+            for (int i = 0; i < 10; i++) {
+                execute(manager, "INSERT INTO tblspace1.tsql (k1,n1 ,s1) values (?,?,?)", Arrays.asList(i, 1, "b"));
+            }
+            execute(manager, "checkpoint TBLSPACE1 -noWait", Collections.emptyList());
+        }
+    }
+
+    @Test
+    public void checkpointCommandNoWaitSuccessTest() throws Exception {
+        List<Long> returnedValues = new ArrayList<>();
+        try (DBManager manager = new DBManager("localhost", new MemoryMetadataStorageManager(), new MemoryDataStorageManager(), new MemoryCommitLogManager(), null, null)) {
+            manager.start();
+            assertTrue(manager.waitForBootOfLocalTablespaces(10000));
+            execute(manager, "CREATE TABLESPACE 'tblspace1'", Collections.emptyList());
+            manager.waitForTablespace("tblspace1", 10000);
+            execute(manager, "CREATE TABLE tblspace1.tsql (k1 string primary key,n1 int,s1 string)", Collections.emptyList());
+            for (int i = 0; i < 10; i++) {
+                execute(manager, "INSERT INTO tblspace1.tsql (k1,n1 ,s1) values (?,?,?)", Arrays.asList(i, 1, "b"));
+            }
+            Thread first = new Thread(() -> {
+                LOGGER.log(Level.INFO, "First thread has started!");
+                returnedValues.add(execute(manager, "checkpoint tblspace1 -noWait", Collections.emptyList()).transactionId);
+                LOGGER.log(Level.INFO, "First thread has finished!");
+            });
+            Thread second = new Thread(() -> {
+                LOGGER.log(Level.INFO, "Second thread has started!");
+                returnedValues.add(execute(manager, "checkpoint tblspace1 -noWait", Collections.emptyList()).transactionId);
+                LOGGER.log(Level.INFO, "Second thread has finished!");
+            });
+            first.start();
+            second.start();
+            first.join();
+            second.join();
+            assertTrue("No checkpoint has finished", returnedValues.contains(0L));
```

Comment:
you are not verifying that the checkpoint was executed.

you could check on the checkpoints metrics

cc @amitvc